### ISSUE TITLE
Investigate the use of prebuilt binaries to simplify installation requirements & experience

### ISSUE THREAD

**Issue by wayneparrott** [RobotWebTools/rclnodejs#865]
Can we simplify the rclnodejs installation requirements? Specifically the compilation for C-lang native api.

To answer this question I have created this tracking issue to investigate the use of [prebuild](https://github.com/prebuild/prebuild) for packaging and distributing compiled binaries.

**Comment by minggangw** [comment]
Of course, we can and could reference [ref-napi](https://github.com/node-ffi-napi/ref-napi/blob/latest/package.json)

**Comment by pabloinigoblasco** [comment]
Is there any example url about how to do this in one "rclnodejs regular package"?

**Comment by minggangw** [comment]
@pabloinigoblasco do you mean you want `rclnodejs` to be a prebuild component in your project?

**Comment by pabloinigoblasco** [comment]
Hello @minggangw 
Well I guess the idea would be having "ready to use debian packages".
When you install a ros package via apt it is IMHO ready to use because all the dependencies are installed and nothing else is required to be done. For example:
```
apt install ros-humble-myowesome-nodejs-package
source /opt/ros/humble/setup.bash
ros2 launch myowesome_nodejs_package example.launch.py
```
The only a ros package developer have to do is usually add the system or ros dependency in the package.xml file.

I have some doubts that would work for nodejs packages (created by rclnodejs-cli).  If this is possible, great, we are fine.

However, I am thinking that the nodejs apt package in ubuntu jammy is 12.22.9 but rclnodejs requires at least 16.x.x , am I right?

If so, I understand that you could install nvm or nodejs manually, etc, but that is not the point.

Do you know if this problem is solved?

If it isnt, I think there could be some "tradeoff solution"  based on some debian package post-install-scripts.
